### PR TITLE
perf(apikey): add apikey/update action

### PR DESCRIPTION
**Pull request by KennethBWSong**
TODO: 1. update schema 2. Skip confirm if only domain changes when local debug 3. add the action to templates

### CODE REVIEW

**Review by SLdragon — DISMISSED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by nliu-ms — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -802,6 +802,11 @@
   "error.core.appIdNotExist": "Cannot find app id: %s. Either your current M365 account does not have permission, or the app has alredy been deleted.",
   "driver.apiKey.description.create": "Create an API key on Developer Portal for authentication in Open API spec.",
   "driver.aadApp.apiKey.title.create": "Creating API key...",
+  "driver.apiKey.description.update": "Update an API key on Developer Portal for authentication in Open API spec.",
+  "driver.aadApp.apiKey.title.update": "Updating API key...",
+  "driver.apiKey.log.skipUpdateApiKey": "Skip updating API key due to same property exists.",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:807]: due to same -> as the same

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -802,6 +802,11 @@
   "error.core.appIdNotExist": "Cannot find app id: %s. Either your current M365 account does not have permission, or the app has alredy been deleted.",
   "driver.apiKey.description.create": "Create an API key on Developer Portal for authentication in Open API spec.",
   "driver.aadApp.apiKey.title.create": "Creating API key...",
+  "driver.apiKey.description.update": "Update an API key on Developer Portal for authentication in Open API spec.",
+  "driver.aadApp.apiKey.title.update": "Updating API key...",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:806]: In title, we don't need ellipses (...) as they show continuation.

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -802,6 +802,11 @@
...
+  "driver.apiKey.log.skipUpdateApiKey": "Skip updating API key due to same property exists.",
+  "driver.apiKey.log.successUpdateApiKey": "Successfully updated API key",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:808]: API key updated successfully!

File: packages/fx-core/src/component/driver/apiKey/interface/updateApiKeyArgs.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+export interface UpdateApiKeyArgs {
+  registrationId: string; // The registration id of the api key
+  name: string; // The name of Api Secret
+  appId: string; // Teams app id
+  apiSpecPath: string; // The location of api spec file
+  applicableToApps?: string; // What app can access the api key. Values can be "SpecificApp" or "AnyApp". Default is "AnyApp".
```
Comment (supkasar) [inline packages/fx-core/src/component/driver/apiKey/interface/updateApiKeyArgs.ts:9]: Which app can access the API key? Values can be "SpecificApp" or "AnyApp". Default is "AnyApp".

File: packages/fx-core/src/component/driver/apiKey/interface/updateApiKeyArgs.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  targetAudience?: string; // What tenant can access the api key. Values can be "HomeTenant" or "AnyTenant". Default is "HomeTenant".
```
Comment (supkasar) [inline packages/fx-core/src/component/driver/apiKey/interface/updateApiKeyArgs.ts:10]: Which tenant can access the API key? Values....

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json]: Fixed, thanks!

File: packages/fx-core/resource/package.nls.json (hunk 2 above)
Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json:806]: Yes it is the message shown in the progress bar.

File: packages/fx-core/resource/package.nls.json (hunk 3 above)
Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json]: Fixed, thanks!

File: packages/fx-core/src/component/driver/apiKey/interface/updateApiKeyArgs.ts (hunk 4 above)
Comment (KennethBWSong) [reply packages/fx-core/src/component/driver/apiKey/interface/updateApiKeyArgs.ts]: Fixed, thanks!

File: packages/fx-core/src/component/driver/apiKey/interface/updateApiKeyArgs.ts (hunk 5 above)
Comment (KennethBWSong) [reply packages/fx-core/src/component/driver/apiKey/interface/updateApiKeyArgs.ts]: Fixed, thanks!

**Review by sffamily — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Review by SLdragon — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**